packages/view/src/components/Statistics/FileIcicleSummary/FileIcicleSummary.tsx

```diff
@@ -0,0 +1,175 @@
+
+  const svg = d3
+    .select($target.current)
+    .attr("viewBox", [0, 0, WIDTH, HEIGHT])
+    .style("font", `${FONT_SIZE}px sans-serif`);
+
+  // Position each partitions
+  const cell = svg
+    .selectAll("g")
+    .data(root.descendants())
+    .join("g")
+    // Hide root node
+    .attr("transform", (d) => `translate(${d.y0 - SINGLE_RECT_WIDTH},${d.x0})`);
+
+  // Draw rect of partition
+  const rect = cell
+    .append("rect")
+    .attr("width", (d) => d.y1 - d.y0 - 1)
+    .attr("height", (d) => rectHeight(d))
+    // directory don't have value field
+    .style(
+      "fill",
+      (d) => COLOR_CODE[d.data.value !== undefined ? "file" : "dir"]
+    )
+    .style("cursor", "pointer");
+
+  // Append labels
+  const text = cell
+    .append("text")
+    .style("user-select", "none")
+    .attr("pointer-events", "none")
+    .attr("x", 4)
+    .attr("y", 13)
+    .attr("fill-opacity", (d) => +labelVisible(d));
+
+  text.append("tspan").text((d) => d.data.name);
+
+  const tspan = text
+    .append("tspan")
+    .attr("fill-opacity", (d) => +labelVisible(d) * 0.7)
+    .text((d) => ` ${d.value?.toLocaleString()}`);
+
+  rect.on("click", (_event, p) => {
+    const positionMap = new WeakMap();
+    // When users click the focused node, change focus to its parent
+    const targetNode = focus === p ? p.parent : p;
+
+    if (targetNode === null) {
+      return;
+    }
+
+    focus = targetNode;
+    const isRootFocused = focus.depth === 0;
+
+    root.each((d) => {
+      positionMap.set(d, {
+        x0: ((d.x0 - targetNode.x0) / (targetNode.x1 - targetNode.x0)) * HEIGHT,
+        x1: ((d.x1 - targetNode.x0) / (targetNode.x1 - targetNode.x0)) * HEIGHT,
+        y0: d.y0 - targetNode.y0 - (isRootFocused ? SINGLE_RECT_WIDTH : 0),
+        y1: d.y1 - targetNode.y0 - (isRootFocused ? SINGLE_RECT_WIDTH : 0),
+      });
+    });
+
+    const t = cell
+      .transition()
+      .duration(750)
+      .attr(
+        "transform",
+        (d) => `translate(${positionMap.get(d).y0},${positionMap.get(d).x0})`
+      );
+
+    rect.transition(t).attr("height", (d) => rectHeight(positionMap.get(d)));
+    text
+      .transition(t)
+      .attr("fill-opacity", (d) => +labelVisible(positionMap.get(d)));
+    tspan
+      .transition(t)
+      .attr("fill-opacity", (d) => +labelVisible(positionMap.get(d)) * 0.7);
+  });
+};
+
+const destroyIcicleTree = ($target: RefObject<SVGSVGElement>) => {
+  d3.select($target.current).selectAll("svg").remove();
+};
+
+const FileIcicleSummary = () => {
+  const [data, setData] = useState<TFileData>();
+  const $summary = useRef<SVGSVGElement>(null);
+
+  useEffect(() => {
+    const getFileData = async () => {
+      const response = await d3.json<TFileData>(FILE_PATH);
+      setData(response);
+    };
+
+    getFileData();
+  }, []);
+
+  useEffect(() => {
+    if (data) {
+      drawIcicleTree($summary, data);
+    }
+
+    // cleanup
+    return () => {
+      destroyIcicleTree($summary);
+    };
```

Comment:
이부분 코드 너무 깔끔하고 좋은 거 같습니다👍👍👍  
저도 비슷하게 리팩토링 해보겠습니다~

Reply:
요 부분 아래와 같이 작성하는 것은 어떨까요? @__@ 
```suggestion
  useEffect(() => {
    if (data) {
      drawIcicleTree($summary, data);
    }

    // cleanup
    return () => destroyIcicleTree($summary);
```

Reply:
```suggestion
  useEffect(() => {
    data && drawIcicleTree($summary, data);

    // cleanup
    return () => destroyIcicleTree($summary);
  }, [data]);
```

요렇게 작성하는 것은 어떻게 생각하세용??